multinet/api/tasks/aql/aql_query.py

```diff
@@ -0,0 +1,30 @@
+from arango.cursor import Cursor
+from arango.exceptions import AQLQueryExecuteError, ArangoServerError
+from celery import shared_task
+
+from multinet.api.models import AqlQuery, Workspace
+from multinet.api.tasks import MultinetCeleryTask
+from multinet.api.utils.arango import ArangoQuery
+
+
+class ExecuteAqlQueryTask(MultinetCeleryTask):
+    task_model = AqlQuery
+
+
+@shared_task(base=ExecuteAqlQueryTask)
+def execute_query(task_id: int) -> None:
+    query_task: AqlQuery = AqlQuery.objects.select_related('workspace').get(id=task_id)
+    workspace: Workspace = query_task.workspace
+    query_str = query_task.query
+
+    try:
+        # Run the query on Arango DB
+        database = workspace.get_arango_db()
+        query = ArangoQuery(database, query_str, time_limit_secs=60)
+        cursor: Cursor = query.execute()
+
+        # Store the results on the task object
+        query_task.query_results = list(cursor)
+        query_task.save()
+    except (AQLQueryExecuteError, ArangoServerError) as err:
+        ExecuteAqlQueryTask.fail_task_with_message(query_task, err.error_message)
```

Comment:
The automatic handling works fine. In fact, be removing my own `try/except`, the task correctly finished with the status `FAILED` (there's another comment on this review regarding that in the tests). Do you have any idea why that is?

Reply:
I understand this now. Since [`MultinetCeleryTask.on_success`](https://github.com/multinet-app/multinet-api/blob/65209c25ae892c32450b4894d9aa08a53dd5198a/multinet/api/tasks/__init__.py#L59-L61) is called once the task finishes _without_ any exceptions, it will mark it as `FINISHED`, even though the task could have been marked as `FAILED`, and not thrown exception. This is a bug in `MultinetCeleryTask`, as that function should check the task status _before_ marking it as finished.

I think for this PR, you could probably just remove this try/except block, if it behaves how it should without it. The fix to `MultinetCeleryTask` could come in a separate PR.

Does that sound good to you?

Reply:
Yeah, that sounds good to me.